src/dataExport.js

```diff
@@ -0,0 +1,134 @@
+﻿import './fileSaver.js';
+
+const snippetUrl = "https://snippet.babylonjs.com";
+
+let currentSnippetToken;
+
+function stringifyShader(name, data, webGPUSupported) {
+    let text = "";
+
+    if (webGPUSupported) {
+        text = "                BABYLON.Effect.ShadersStoreWGSL[\"" + name + "\"]=";
+    } else {
+        text = "                BABYLON.Effect.ShadersStore[\"" + name + "\"]=";
+    }
+
+    let splits = data.split("\n");
+    for (let index = 0; index < splits.length; index++) {
+
+        if (splits[index] !== "") {
+            text += "                \"" + splits[index] + "\\r\\n\"";
+
+            if (index != splits.length - 1) {
+                text += "+\r\n";
+            } else {
+                text += ";\r\n";
+            }
+        } else {
+            text += "\r\n";
+        }
+    }
+
+    return text;
+}
+
+function addContentToZip(zip, name, url, replace, buffer, then) {

```

Comment:
a perfect candidate for fetch and promise architecture? (async using `fetch`?)

Reply:
Will look into it over the weekend.

Reply:
Finally got around refactoring this.